### PR TITLE
add legacy field selector for names

### PR DESCRIPTION
**Pull request by deads2k**
I think this gets me down the dozen or so "real" field selectors I need to convert for groupification.

### CODE REVIEW

**Comment by deads2k** [comment]
@enj since you like removing boiilerplate

File: pkg/deploy/apis/apps/v1/conversion.go [hunk 1]
```diff
@@ -119,10 +118,5 @@ func addConversionFuncs(scheme *runtime.Scheme) error {
 		return err
 	}
 
-	if err := scheme.AddFieldLabelConversionFunc("v1", "DeploymentConfig",
-		oapi.GetFieldLabelConversionFunc(newer.DeploymentConfigToSelectableFields(&newer.DeploymentConfig{}), nil),
-	); err != nil {
-		return err
-	}
```
Comment (ironcladlou) [inline pkg/deploy/apis/apps/v1/conversion.go]: Why is this required [for buildconfigs](https://github.com/openshift/origin/pull/15963/files#diff-c6033988686cf6611cf7e928d8dbea2aR178) but not here?

File: pkg/image/apis/image/v1/conversion.go [hunk 2]
```diff
@@ -280,12 +280,6 @@ func addConversionFuncs(scheme *runtime.Scheme) error {
 		return err
 	}
 
-	if err := scheme.AddFieldLabelConversionFunc("v1", "Image",
-		oapi.GetFieldLabelConversionFunc(newer.ImageToSelectableFields(&newer.Image{}), nil),
-	); err != nil {
-		return err
-	}
-
```
Comment (ironcladlou) [inline pkg/image/apis/image/v1/conversion.go]: Why is this required [for buildconfigs](https://github.com/openshift/origin/pull/15963/files#diff-c6033988686cf6611cf7e928d8dbea2aR178) but not here?

File: pkg/image/apis/image/v1/conversion.go [hunk 3]
```diff
@@ -280,12 +280,6 @@ func addConversionFuncs(scheme *runtime.Scheme) error {
...
 	if err := scheme.AddFieldLabelConversionFunc("v1", "ImageStream",
 		oapi.GetFieldLabelConversionFunc(newer.ImageStreamToSelectableFields(&newer.ImageStream{}), map[string]string{"name": "metadata.name"}),
```
Comment (ironcladlou) [inline pkg/image/apis/image/v1/conversion.go]: Missing port to the new `LegacyMetaV1FieldSelectorConversionWithName `?

File: pkg/oauth/apis/oauth/v1/conversion.go [hunk 4]
```diff
@@ -20,12 +20,6 @@ func addConversionFuncs(scheme *runtime.Scheme) error {
 		return err
 	}
 
-	if err := scheme.AddFieldLabelConversionFunc("v1", "OAuthClient",
-		oapi.GetFieldLabelConversionFunc(oauthapi.OAuthClientToSelectableFields(&oauthapi.OAuthClient{}), nil),
-	); err != nil {
-		return err
-	}
-
```
Comment (ironcladlou) [inline pkg/oauth/apis/oauth/v1/conversion.go]: Why is this required [for buildconfigs](https://github.com/openshift/origin/pull/15963/files#diff-c6033988686cf6611cf7e928d8dbea2aR178) but not here?

File: pkg/user/apis/user/v1/conversion.go [hunk 5]
```diff
@@ -8,22 +8,11 @@ import (
 )
 
 func addConversionFuncs(scheme *runtime.Scheme) error {
-	if err := scheme.AddFieldLabelConversionFunc("v1", "Group",
-		oapi.GetFieldLabelConversionFunc(userapi.GroupToSelectableFields(&userapi.Group{}), nil),
-	); err != nil {
-		return err
-	}
-
```
Comment (ironcladlou) [inline pkg/user/apis/user/v1/conversion.go]: Why is this required [for buildconfigs](https://github.com/openshift/origin/pull/15963/files#diff-c6033988686cf6611cf7e928d8dbea2aR178) but not here?

File: pkg/user/apis/user/v1/conversion.go [hunk 6]
```diff
@@ -8,22 +8,11 @@ import (
...
 		oapi.GetFieldLabelConversionFunc(userapi.IdentityToSelectableFields(&userapi.Identity{}), nil),
 	); err != nil {
 		return err
 	}
 
-	if err := scheme.AddFieldLabelConversionFunc("v1", "User",
-		oapi.GetFieldLabelConversionFunc(userapi.UserToSelectableFields(&userapi.User{}), nil),
-	); err != nil {
-		return err
-	}
```
Comment (ironcladlou) [inline pkg/user/apis/user/v1/conversion.go]: Why is this required [for buildconfigs](https://github.com/openshift/origin/pull/15963/files#diff-c6033988686cf6611cf7e928d8dbea2aR178) but not here?

File: pkg/deploy/apis/apps/v1/conversion.go (hunk 1 above)
Comment (deads2k) [inline pkg/deploy/apis/apps/v1/conversion.go]: This doesn't have a special case to map "name" to "metadata.name" for field selection.

File: pkg/image/apis/image/v1/conversion.go (hunk 2 above)
Comment (deads2k) [inline pkg/image/apis/image/v1/conversion.go]: doesn't have name -> metadata.name

File: pkg/image/apis/image/v1/conversion.go (hunk 3 above)
Comment (deads2k) [inline pkg/image/apis/image/v1/conversion.go]: > Missing port to the new LegacyMetaV1FieldSelectorConversionWithName?

didn't forget this one. This field selector actually lists many fields so the conversion can't be done via a straight default.  It will require more thought and code.

File: pkg/oauth/apis/oauth/v1/conversion.go (hunk 4 above)
Comment (deads2k) [inline pkg/oauth/apis/oauth/v1/conversion.go]: doesn't do name->metadata.name

File: pkg/user/apis/user/v1/conversion.go (hunk 5 above)
Comment (deads2k) [inline pkg/user/apis/user/v1/conversion.go]: doesn't have name->metadata.name

File: pkg/user/apis/user/v1/conversion.go (hunk 6 above)
Comment (deads2k) [inline pkg/user/apis/user/v1/conversion.go]: doesn't have name->metadata.name

**Comment by ironcladlou** [comment]
That add/remove ratio...

/lgtm
